### PR TITLE
Re-add netmiko send_config

### PR DESCRIPTION
**Pull request by ktbyers**
Re-submit https://github.com/napalm-automation/brigade/pull/67 to fix rebase issue and to add unit tests.

### CODE REVIEW

**Comment by ktbyers** [comment]
Here is example test code:

```
from brigade.core import Brigade
from brigade.plugins.inventory.simple import SimpleInventory
from brigade.plugins.tasks.connections import netmiko_connection
from brigade.plugins.tasks.networking import netmiko_send_config


def result_printer(result):
    for host, result_obj in result.items():
        print('-' * 80)
        print(result_obj[0].result)
        print('-' * 80)


brigade = Brigade(inventory=SimpleInventory("hosts.yml", "groups.yml"), dry_run=False)
b = brigade.filter(group='cisco-ios')

# Explicitly create the connection and execute config changes
b.run(task=netmiko_connection, num_workers=60)
commands = ['logging buffer 30000', 'no logging console']
result = b.run(netmiko_send_config, num_workers=60, config_commands=commands)
result_printer(result)

# Explicitly create the connection and execute config changes
brigade2 = Brigade(inventory=SimpleInventory("hosts.yml", "groups.yml"), dry_run=False)
b2 = brigade2.filter(group='cisco-ios')
result = b2.run(netmiko_send_config, num_workers=60, config_file="config_load.txt")
result_printer(result)
```

**Comment by ktbyers** [comment]
Here is the output from running that code:

```
python test_netmiko_config2.py 
--------------------------------------------------------------------------------
config term
Enter configuration commands, one per line.  End with CNTL/Z.
pynet-rtr1(config)#logging buffer 30000
pynet-rtr1(config)#no logging console
pynet-rtr1(config)#end
pynet-rtr1#
--------------------------------------------------------------------------------
--------------------------------------------------------------------------------
config term
Enter configuration commands, one per line.  End with CNTL/Z.
pynet-rtr2(config)#logging buffer 30000
pynet-rtr2(config)#no logging console
pynet-rtr2(config)#end
pynet-rtr2#
--------------------------------------------------------------------------------
--------------------------------------------------------------------------------
config term
Enter configuration commands, one per line.  End with CNTL/Z.
pynet-rtr1(config)#logging buffered 10000
pynet-rtr1(config)#logging console
pynet-rtr1(config)#end
pynet-rtr1#
--------------------------------------------------------------------------------
--------------------------------------------------------------------------------
config term
Enter configuration commands, one per line.  End with CNTL/Z.
pynet-rtr2(config)#logging buffered 10000
pynet-rtr2(config)#logging console
pynet-rtr2(config)#end
pynet-rtr2#
--------------------------------------------------------------------------------
```

File: brigade/plugins/tasks/networking/netmiko_send_config.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    """
+    Execute Netmiko send_config_set method (or send_config_from_file)
+    Arguments:
+        config_commands(list, optional): Commands to configure on the remote network device.
+        config_file(str, optional): File to read configuration commands from.
+
+        kwargs (dict, optional): Additional arguments to pass to method.
+    Returns:
+        :obj:`brigade.core.task.Result`:
+          * result (``dict``): dictionary with the result of the getter
```
Comment (dbarrosop) [inline brigade/plugins/tasks/networking/netmiko_send_config.py]: I guess this is just due to copy and paste :)

File: brigade/plugins/tasks/networking/netmiko_send_config.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+from brigade.core.task import Result
+
+
+def netmiko_send_config(task, config_commands=None, config_file=None, **kwargs):
+    """
+    Execute Netmiko send_config_set method (or send_config_from_file)
+    Arguments:
+        config_commands(list, optional): Commands to configure on the remote network device.
+        config_file(str, optional): File to read configuration commands from.
+
```
Comment (dbarrosop) [inline brigade/plugins/tasks/networking/netmiko_send_config.py]: Probably remove the space, not sure if sphinx will interpret it as a new section

**Comment by dbarrosop** [comment]
This is really nice. Just a couple of nitpicky comments. Also, have you tested if sphinx renders the documentation? I think it should.

**Comment by ktbyers** [comment]
@dbarrosop Okay, I think I fixed your two items. Not sure about the second one, but I copied the spacing pattern you had used for a napalm plugin.

Sphinx is all fubar on Fedora (or more exactly pandoc is fubar requiring a Haskell compiler). I will have to figure out a way to look at it. I looked after the fact at the readthedocs (on the other Netmiko task) and it looks like it renders properly.

**Comment by dbarrosop** [comment]
> but I copied the spacing pattern you had used for a napalm plugin

Ok, I will probably have to fix that. There is no reason for that space and I am afraid sphinx might interpret that as a new section.

> Sphinx is all fubar on Fedora (or more exactly pandoc is fubar requiring a Haskell compiler).

If it's useful we can add a docker container.

**Pull request merged**